app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -4,6 +4,7 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
 
   def github
     @user = User.from_omniauth(auth)
+    update_users_avatar?
```

Comment:
The responsibility of `update_user_avatar` is just to update the avatar, so I suggest to replace the method invocation with

```ruby
update_user_avatar if avatar_needs_updated?
```

```ruby
def update_users_avatar
  @user.update_avatar(github_avatar) 
end
```